lib/qiita/markdown/filters/external_link.rb

```diff
@@ -0,0 +1,35 @@
+module Qiita
+  module Markdown
+    module Filters
+      class ExternalLink < HTML::Pipeline::Filter
+        def call
+          return doc unless hostname
+
+          doc.search("a").each do |anchor|
+            next if anchor["href"].nil?
+            href = anchor["href"].strip
+            href_host = host_of(href)
+            next unless href_host
+            unless href_host.end_with?(hostname)
```

Comment:
Thank you for the comment. Yes, it is not the expected behaviour. I modify its code.